### PR TITLE
Add var for MaxAuthTries

### PR DESCRIPTION
**Pull request by maximiliend**
This add a variable which allow to define the maximum number of authentication.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -48,6 +48,7 @@
   $sshd_config_allowusers              = [],
   $sshd_config_denygroups              = [],
   $sshd_config_denyusers               = [],
+  $sshd_config_maxauthtries            = 5,
```
Comment (ghoneycutt) [inline manifests/init.pp]: It appears the default was 6, so that should remain

File: templates/sshd_config.erb [hunk 2]
```diff
@@ -61,7 +61,10 @@ PermitRootLogin <%= @permit_root_login %>
 <% if @sshd_config_strictmodes -%>
 StrictModes <%= @sshd_config_strictmodes %>
 <% end -%>
-#MaxAuthTries 6
```
Comment (ghoneycutt) [inline templates/sshd_config.erb]: We keep the default commented out to aid in debugging

**Comment by ghoneycutt** [comment]
Thanks @maximiliend 

Could you please document the new parameter in the README? We also need spec tests in order to accept. Look at a similar param and copy how it is tested.

**Comment by maximiliend** [comment]
Thank you for your advice, the defaults values are amended and the README enhanced. I'll look for specific tests.

File: spec/classes/init_spec.rb [hunk 3]
```diff
@@ -608,6 +608,7 @@
     it { should contain_file('sshd_config').with_content(/^AcceptEnv L.*$/) }
     it { should_not contain_file('sshd_config').with_content(/^AuthorizedKeysFile/) }
     it { should_not contain_file('sshd_config').with_content(/^StrictModes/) }
+    it { should contain_file('sshd_config').with_content(/^MaxAuthTries/) }
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: This should have a `6` in it, as by default the line should have that

File: spec/classes/init_spec.rb [hunk 4]
```diff
@@ -2045,6 +2047,25 @@
     end
   end
 
+  context 'with sshd_config_maxauthtries not being a valid number' do
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: To get this to work, we need to validate the parameter in the ssh class. Something like

`validate_integer()`

File: manifests/init.pp [hunk 5]
```diff
@@ -48,6 +48,7 @@
   $sshd_config_allowusers              = [],
   $sshd_config_denygroups              = [],
   $sshd_config_denyusers               = [],
+  $sshd_config_maxauthtries            = 6,
```
Comment (ghoneycutt) [inline manifests/init.pp]: Sometimes people will inadvertently use a string `'6'`. To deal with this we can do something like

``` puppet
$sshd_config_maxauthtries_real = $sshd_config_maxauthtries + 0
validate_integer($sshd_config_maxauthtries_real)
```

File: spec/classes/init_spec.rb (hunk 3 above)
Comment (maximiliend) [inline spec/classes/init_spec.rb]: I understood ;)

File: spec/classes/init_spec.rb (hunk 4 above)
Comment (maximiliend) [inline spec/classes/init_spec.rb]: Ok or maybe something like :
validate_re($sshd_config_maxauthtries, '^\d+$', "ssh::sshd_config_maxauthtries must be a valid number and is set to <${sshd_config_maxauthtries}>.")
Thanks you.

File: manifests/init.pp (hunk 5 above)
Comment (maximiliend) [inline manifests/init.pp]: Got it too ! I'll use directly '6'. But thanks for the advice.

Thanks a lot for your help. Sorry for the mistakes that I made. Tell me if there is something else.

I'll push again my commit.

File: spec/classes/init_spec.rb (hunk 4 above)
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: validate_re() expects a string and it really should be an integer

File: spec/classes/init_spec.rb [hunk 6]
```diff
@@ -608,6 +608,7 @@
     it { should contain_file('sshd_config').with_content(/^AcceptEnv L.*$/) }
     it { should_not contain_file('sshd_config').with_content(/^AuthorizedKeysFile/) }
     it { should_not contain_file('sshd_config').with_content(/^StrictModes/) }
+    it { should contain_file('sshd_config').with_content(/^MaxAuthTries 6/) }
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: There are a handful of other spots that a line similar to this one should be in.

**Comment by maximiliend** [comment]
I don't know if I can add more spec tests. I hope this is enough.

**Comment by maximiliend** [comment]
Is there anything else missing ?

**Comment by jthiesfeld** [comment]
What is the status of this patch as it would be useful in my environment as well?

**Comment by ghoneycutt** [comment]
@jthiesfeld the spec tests were refactored and so this cannot be merged due to that. You want to take a crack at it?

**Comment by jthiesfeld** [comment]
I'm new to puppet, ruby, and git.

It looks like maximiliend has a few pull requests, but just the rspec
change didn't get accepted?  Any pointers on how to get started?

**Pull request closed without merging**